src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDeployment.java

```diff
@@ -52,15 +52,25 @@ public void deploy(DeployConfiguration config) throws AppEngineException {
     Preconditions.checkNotNull(config);
     Preconditions.checkNotNull(config.getDeployables());
     Preconditions.checkArgument(config.getDeployables().size() > 0);
+    File workingDirectory = null;
 
     List<String> arguments = new ArrayList<>();
     arguments.add("deploy");
-    for (File deployable : config.getDeployables()) {
-      if (!deployable.exists()) {
-        throw new IllegalArgumentException(
-            "Deployable " + deployable.toPath() + " does not exist.");
+
+    // Unfortunately, 'gcloud app deploy' does not let you pass a staging directory as a deployable.
+    // Instead, we have to run 'gcloud app deploy' from the staging directory to achieve this.
+    // So, if we find that the only deployable in the list is a directory, we just run the command
+    // from that directory without passing in any deployables to gcloud.
+    if (config.getDeployables().size() == 1 && config.getDeployables().get(0).isDirectory()) {
```

Comment:
Meh, I don't feel very strongly about this. I wanted to avoid having this complexity here. But, by generating app.yamls, we'd be adding complexity somewhere else, so I guess I'm fine